nulib/meadow
# Add streaming to dev environment via MediaConvert.Mock

## Description

This PR adds the ability to stream audio/video content directly from minio in the developer environment. Instead of sending the media to MediaConvert for transcoding, it simply copies the source file to the streaming bucket and returns the location of the file as the playlist. Minio can serve them to the front end via HTTP.

### Testing

1. Create a work with an audio file set and a video file set.
2. Save the following to a file called `streaming_test.html` somewhere outside your Meadow directory.

    ```html
    <html>
      <head>
        <title>Streaming Test</title>
      </head>
      <body>
        <p>
          <video controls>
            <source src=""/>
          </video>
        </p>
        <p>
          <audio controls>
            <source src=""/>
          </audio>
        </p>
      </body>
    </html>
    ```

3. Request the work in graphiql and get its file sets' `streamingUrl` properties.
4. Paste the URL for the video into the video source and the audio into the audio source above.
5. Load `streaming_test.html` in a browser.

You should have fully functioning video and audio players in the browser window.

![image](https://user-images.githubusercontent.com/196872/120719810-8d4e8480-c490-11eb-8fb6-9bba4e2f8806.png)


### Notes
* In order to work with different media types, the front end either needs to match the `audio`/`video` tag's `type` attribute to the file extension of the URL, or just leave out the `type` attribute altogether and let the browser figure it out.
* In order to make streaming work with a Minio bucket, the `Config.streaming_host()` function needed to be replaced with `Config.streaming_url()` to return the entire URL stem. All downstream functions have been updated to account for this, but the Terraform change in this PR will have to be applied after merge.
